source/MaterialXGenShader/ColorManagementSystem.cpp

```diff
@@ -0,0 +1,94 @@
+#include <MaterialXGenShader/ColorManagementSystem.h>
+
+#include <MaterialXGenShader/ShaderGenerator.h>
+#include <MaterialXGenShader/Nodes/SourceCodeNode.h>
+
+namespace MaterialX
+{
+
+ColorSpaceTransform::ColorSpaceTransform(const string& ss, const string& ts, const TypeDesc* t)
+    : sourceSpace(ss)
+    , targetSpace(ts)
+    , type(t)
+{
+    if (type != Type::COLOR3 && type != Type::COLOR4)
+    {
+        throw ExceptionShaderGenError("Color space transform can only be a color3 or color4.");
+    }
+}
+
+
+ColorManagementSystem::ColorManagementSystem(ShaderGenerator& shadergen, const string& configFile)
+    : _configFile(configFile)
+    , _shadergen(shadergen)
+{
+}
+
+bool ColorManagementSystem::registerImplementation(const ColorSpaceTransform& transform, CreatorFunction<ShaderNodeImpl> creator)
+{
+    if (transform.type)
```

Comment:
Maybe remove this check, since type cannot be null. This method could then have void return type.